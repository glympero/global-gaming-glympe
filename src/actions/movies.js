//SET_MOVIES
const setMovie = (movies) => ({
    type: 'SET_MOVIES',
    movies
  });

// REMOVE_MOVIE
const removeMovie = ({ id } = {}) => ({
    type: 'REMOVE_MOVIE',
    id
});

// EDIT_MOVIE
const editMovie = (id, updates) => ({
    type: 'EDIT_MOVIE',
    id,
    updates
});

export { setMovie, removeMovie, editMovie };