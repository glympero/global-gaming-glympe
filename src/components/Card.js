import React from 'react';
import { Link } from 'react-router-dom';

class Clock extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            day: '',
            date: '',
            month: '',
            year: '',
            hours: '',
            minutes: '',
            sec: '',
            isMounted: false
        }
        let refreshIntervalId;
    }

    getDay = (d) => {
        let day = d.getDay();
        let dayarr =["Sun","Mon","Tues","Wed","Thurs","Fri","Sat"];
        day = dayarr[day];
        return day;
    }

    getDate = (d) => {
        return d.getDate();
    }

    getMonth = (d) => {
        let month = d.getMonth();
        let montharr =["Jan","Feb","Mar","April","May","June","July","Aug","Sep","Oct","Nov","Dec"];
        month = montharr[month];
        return month;
    }

    getYear = (d) => {
        return  d.getFullYear();
    }

    getHours = (d) => {
        return d.getHours();
    }

    getMinutes = (d) => {
        if(d.getMinutes() < 10){
            return `0${d.getMinutes()}`;
        }
        return `${d.getMinutes()}`;
    }

    getSeconds = (d) => {
        if(d.getSeconds() < 10){
            return `0${d.getSeconds()}`;
        }
        return d.getSeconds();
    }
    clock = () => {
        const d = new Date();
        this.setState(() => ({
            day: this.getDay(d),
            date: this.getDate(d),
            month: this.getMonth(d),
            year: this.getYear(d),
            hours: this.getHours(d),
            minutes: this.getMinutes(d),
            sec: this.getSeconds(d)
        }));
      }

      componentDidMount =() => {
        this.refreshIntervalId = setInterval(this.clock, 1000);
      }

    componentWillUnmount = () => {
        clearInterval(this.refreshIntervalId);
    }

    render() {
        return (
            <div className="blog-card spring-fever">
                <Link to='/movies'>
                    <div className="title-content">
                        <h2>Movies List</h2>
                        <hr />
                        <div className="intro">Click Here For Movies List</div>
                    </div>
                    <div className="card-info">Show Movies</div>
                    <div className="utility-info">
                        <ul className="utility-list">
                            <li className="date">{this.state.day} {this.state.date} {this.state.month} {this.state.year}</li>
                            {this.state.hours && <li className="comments">{this.state.hours} : {this.state.minutes} : {this.state.sec}</li>}
                        </ul>
                    </div>
                    <div className="gradient-overlay"></div>
                     <div className="color-overlay"></div>
                </Link>
            </div>
            
        );
    }
};

export default Clock;