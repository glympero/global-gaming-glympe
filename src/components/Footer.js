import React from 'react';

const Footer = () => (
    <div className="footer">
      <div id="button"></div>
      <div id="container">
        <div id="cont">
          <div className="footer_center">
            <div className="footer-social-icons">
              <ul className="social-icons">
                  <li><a href="http://glympero-portfolio.herokuapp.com/" target="_blank" rel="noopener noreferrer" className="social-icon"> <i className="fa fa-globe"></i></a></li>
                  <li><a href="https://twitter.com/lympero" target="_blank" rel="noopener noreferrer" className="social-icon"> <i className="fa fa-twitter"></i></a></li>
                  <li><a href="https://github.com/glympero" target="_blank" rel="noopener noreferrer" className="social-icon"> <i className="fa fa-github"></i></a></li>
                  <li><a href="https://www.linkedin.com/in/george-lymperopoulos-398a6551" target="_blank" rel="noopener noreferrer" className="social-icon"> <i className="fa fa-linkedin"></i></a></li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
);

export default Footer;
