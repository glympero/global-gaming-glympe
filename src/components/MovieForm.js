import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

class MovieForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: props.movie ? props.movie.title : '',
      genre: props.movie ? props.movie.genre : '',
      directorName: props.movie ? props.movie.director.name : '',
      directorGender: props.movie ? props.movie.director.gender : '',
      release_date: props.movie ? moment(props.movie.release_date, 'DD MMMM YYYY') : '',
      error: ''
    }
  }

  onTitleChange = (e) => {
    const title = e.target.value;
    this.setState(() => ({
      title
    }));
  }
  
  onSubmit = (e) => {
    e.preventDefault();
    if(
      !this.state.title || 
      !this.state.genre ||
      !this.state.directorName || 
      !this.state.directorGender || 
      !this.state.release_date
    ) {
      this.setState(() => ({
        error: 'All fields are required'
      }));
    }else {
      this.setState(() => ({
        error: ''
      }));
      const title = this.state.title;
      const genre = this.state.genre;
      const directorName = this.state.directorName;
      const directorGender = this.state.directorGender;
      const release_date = this.state.release_date;
      this.props.onSubmit({ 
        title,
        genre,
        directorName,
        directorGender,
        release_date
      });
    }
  }

  render() {
    return (
      <form className='form' onSubmit={this.onSubmit}>
        { !!this.state.error && <p className='form__error'>{this.state.error}</p>}
        <input className='text-input' type='text' value={this.state.title} onChange={this.onTitleChange} placeholder='title' autoFocus/>
        <input className='text-input' type='text' value={this.state.genre} disabled placeholder='gender' />
        <input className='text-input' type='text' value={this.state.directorName} disabled placeholder='director name' />
        <input className='text-input' type='text' value={this.state.directorGender} disabled placeholder='director gender'/>
        <input className='text-input' type='text' value={this.state.release_date} disabled placeholder='release date' />
        <button className='button'>Edit Title / Close</button>
      </form>
    );
  }
}

MovieForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  movie: PropTypes.object.isRequired
};

export default MovieForm;