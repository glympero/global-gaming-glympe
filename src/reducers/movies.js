const moviesReducerDefaultState = [];

const moviesReducer = (state = moviesReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_MOVIES':
            return action.movies;
        case 'REMOVE_MOVIE':
            return state.filter(movie => movie.id !== action.id);
        case 'EDIT_MOVIE':
            return state.map(movie => {
                if(movie.id === action.id) {
                    return {
                        ...movie,
                        ...action.updates
                    }
                }
                return movie;
            })
        default:
            return state;
    }
};

export default moviesReducer;