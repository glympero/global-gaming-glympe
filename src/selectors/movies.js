const getVisibleMovies = (movies, filters) => {
    const {text, sortBy, genre } = filters;
    return movies.filter(movie => {
        const textMatched = typeof text === 'string' && movie.title.toLowerCase().indexOf(text.toLowerCase()) !== -1;
        const genreMatched = typeof genre === 'string' && genre === 'all' ? true : movie.genre.toLowerCase().indexOf(genre.toLowerCase()) !== -1;
        return textMatched && genreMatched;
    }).sort((a, b) => {
        if(sortBy === 'title') {
            return a.title > b.title ? 1 : -1;
        }else if (sortBy === 'director') {
            return a.director.toLowerCase() > b.director.toLowerCase() ? 1 : -1;
        }
    });
};

export default getVisibleMovies;