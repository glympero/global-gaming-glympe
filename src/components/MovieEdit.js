import React from 'react';
import PropTypes from 'prop-types';
import MovieModal from './MovieModal';
import { connect } from 'react-redux';
import { editMovie, removeMovie } from '../actions/movies';

export class MovieEdit extends React.Component {
  
  onRemove = () => {
    this.props.removeMovie( {id: this.props.movie.id} );
    this.props.onClose();
  }

  onSubmit = (movie) => {
    this.props.editMovie(this.props.movie.id, movie);
    this.props.onClose();
  }

  render() {
    return (
        <MovieModal
          onSubmit={this.onSubmit}
          movie={this.props.movie}
          remove={this.onRemove}
        />
    )
  }
}

const mapStateToProps = (state, props) => {
  let movie = state.movies.find( e => e.id === props.id);
  return {
    movie
  }
};

const mapDispatchToProps = (dispatch, props) => {
  return {
    removeMovie: (data) => {
      dispatch(removeMovie(data));
    },
    editMovie: (id, movie) => {
      dispatch(editMovie(id, movie));
    }
  }
};

MovieEdit.propTypes = {
  removeMovie: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired,
  editMovie: PropTypes.func.isRequired,
  movie: PropTypes.object
};

export default connect(mapStateToProps, mapDispatchToProps)(MovieEdit);