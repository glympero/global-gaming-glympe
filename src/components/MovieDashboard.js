import React from 'react';
import Title from './Title';
import MovieList from './MovieList';
import MovieListFilters from './MovieListFilters';
import Card from './Card';

const MovieDashboard = (props) => {

    return (
      <div>
        <Title title="Welcome to our Movies database" />
        <Card />
    </div>
    );
  };

export default MovieDashboard;