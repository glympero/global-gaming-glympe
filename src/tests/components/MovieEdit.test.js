import React from 'react';
import { shallow } from 'enzyme';
import { MovieEdit } from '../../components/MovieEdit';
import movies from '../../fixtures/data';

let editMovie, removeMovie, onClose, wrapper, match
beforeEach(() => {
    editMovie = jest.fn();
    removeMovie = jest.fn();
    onClose = jest.fn();
    match = { params: 1 }

    wrapper = shallow(
        <MovieEdit 
            editMovie={editMovie} 
            removeMovie={removeMovie} 
            onClose={onClose} 
            match={match}
            movie={movies[0]}
        />
    )
});

test('should render EditMoviePage correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should test editMovie correctly', () => {
    wrapper.find('MovieForm').prop('onSubmit')(movies[1]);
    expect(onClose).toHaveBeenLastCalledWith();
    expect(editMovie).toHaveBeenLastCalledWith(movies[0].id, movies[1]);
});

test('should test remove movie correctly', () => {
    wrapper.find('button').prop('onClick')();
    expect(onClose).toHaveBeenLastCalledWith();
    expect(removeMovie).toHaveBeenLastCalledWith({ id: movies[0].id})
});