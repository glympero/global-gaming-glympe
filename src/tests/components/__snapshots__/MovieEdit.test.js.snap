// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditMoviePage correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Absolute Bite
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <MovieForm
      movie={
        Object {
          "director": Object {
            "gender": "Female",
            "name": "Walsh Mertie",
          },
          "genre": "Drama@april.biz",
          "id": "1",
          "release_date": "12 June 2017",
          "title": "Absolute Bite",
        }
      }
      onSubmit={[Function]}
    />
    <button
      className="button button--secondary"
      onClick={[Function]}
    >
      Remove
    </button>
  </div>
</div>
`;
